cvem/connectors/one/OpenNebula.py

```diff
@@ -148,10 +148,10 @@ def get_vm_list():
 			res = []
 			for vm in res_vm.VM:
 				host = HostInfo(int(vm.HISTORY_RECORDS.HISTORY[0].HID), vm.HISTORY_RECORDS.HISTORY[0].HOSTNAME)
-				new_vm = VirtualMachineInfo(int(vm.ID), host, vm.TEMPLATE.MEMORY * 1024, vm)
+				new_vm = VirtualMachineInfo(int(vm.ID), host, int(vm.TEMPLATE.MEMORY) * 1024, vm)
 				new_vm.user_id = vm.UID
 				if vm.USER_TEMPLATE.MEM_TOTAL:
-					new_vm.set_memory_values(int(vm.USER_TEMPLATE.MEM_TOTAL_REAL),
+					new_vm.set_memory_values(int(vm.REALMEMORY),
```

Comment:
Hi, some cut from CPMInfo.py

		self.real_memory =  None
		""" Real memory assigned to the VM """
		self.total_memory =  None
		""" Total memory reported by the S.O. (less than real_memory) """
		self.free_memory =  None
		""" Free memory of the VM """
		self.allocated_memory = allocated_memory
		""" Amount of memory originally allocated by the CMP """

So real memory have to be equal to virsh dominfo "Used memory ", that it is that memory, which we are set by virsh setmem command. So this is absolutely right.

Your onegate_published script reports vm.USER_TEMPLATE.MEM_TOTAL_REAL as value which is almost some as allocated_memory